web/src/features/charts/graphUtils.ts

```diff
@@ -74,34 +74,34 @@ export const getGenerationTypeKey = (name: string): GenerationType | undefined =
   return undefined;
 };
 
-export function getTotalElectricity(zoneData: ZoneDetail, mixMode: Mode) {
-  const isConsumption = mixMode === Mode.CONSUMPTION;
-  if (isConsumption) {
-    if (zoneData.totalConsumption === null) {
-      return Number.NaN;
-    }
-    return zoneData.totalConsumption;
-  }
+/** Returns the total electricity that is available in the zone (e.g. production + discharge + imports) */
+export function getTotalElectricityAvailable(zoneData: ZoneDetail, mixMode: Mode) {
+  const includeImports = mixMode === Mode.CONSUMPTION;
+  const totalDischarge = zoneData.totalDischarge ?? 0;
+  const totalImport = zoneData.totalImport ?? 0;
+
   if (zoneData.totalProduction === null) {
     return Number.NaN;
   }
-  // Electricity: Handle discharge case if available, else default to production
-  return zoneData.totalProduction + (zoneData.totalDischarge ?? 0);
+
+  return zoneData.totalProduction + totalDischarge + (includeImports ? totalImport : 0);
 }
 
-export function getTotalEmissions(zoneData: ZoneDetail, mixMode: Mode) {
-  const isConsumption = mixMode === Mode.CONSUMPTION;
-  if (isConsumption) {
-    if (zoneData.totalCo2Consumption === null) {
-      return Number.NaN;
-    }
-    return zoneData.totalCo2Consumption;
-  }
+/** Returns the total emissions that is available in the zone (e.g. production + discharge + imports) */
+export function getTotalEmissionsAvailable(zoneData: ZoneDetail, mixMode: Mode) {
```

Comment:
While this function is pretty simple and easy to understand it is almost identical to the one above. Do you think it makes sense to try and combine them?

Reply:
Haha, it used to be one, but we decided to split it as it was simply too confusing when using it to have one function for both use cases :D Also switching between the two "types" with an argument became harder to understand and reason about.